src/core/utils.js

```diff
@@ -102,3 +104,16 @@ export function loadAddonIfNeeded(
   return fetchAddon({ slug, api: state.api })
     .then(({ entities }) => dispatch(loadEntities(entities)));
 }
+
+export function isAllowedOrigin(urlString, { allowedOrigins = [config.get('amoCDN')] } = {}) {
+  let isValid = false;
+  try {
+    const parsedURL = url.parse(urlString);
+    if (allowedOrigins.includes(`${parsedURL.protocol}//${parsedURL.host}`)) {
+      isValid = true;
+    }
+  } catch (e) {
+    log.error('invalid urlString provided');
```

Comment:
This feels fairly generic. Can you include the `urlString` in here? It might be good to limit the catch to something like:

```js
let parsedURL;
try {
  parsedURL = url.parse(urlString);
} catch (e) {
  log.error(`invalid urlString provided to isAllowedOrigin: ${urlString}`);
  return false;
}
return allowedOrigins.includes(`${parsedURL.protocol}//${parsedURL.host}`));